docs/proposals/images/machinepool-machines/inframachinepool-create-replicas.plantuml

```diff
@@ -0,0 +1,17 @@
+@startuml inframachinepool-create-replicas
+:InfraMachinePool create replicas reconciliation;
+if (status.InfrastructureMachineKind is set on InfraMachinePool) then (no)
+  :set status.InfrastructureMachineKind to InfraMachinePoolMachine;
+else (yes)
+endif
+repeat 
+  :create provider specific resource representing a replica;
+repeat while (~# replicas < # desired replicas ?) is (yes)
```

Comment:
this should probably go before "create provider specific resource representing a replica", we don't create a first one no matter what

Reply:
That makes sense, let me try to restructure the loop condition.